Migrate account list to new user model

ezaquarii opened the pull request:
Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

ezaquarii commented:
@tobiasKaminsky Linter errors were there before me.

ezaquarii commented on src/main/java/com/owncloud/android/ui/dialog/MultipleAccountsDialog.java:
```diff
@@ -117,21 +120,21 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
+        List<User> users = accountManager.getAllUsers();
+        List<UserListItem> adapterUserList = new ArrayList<>(users.size());
+        for (User user : users) {
+            adapterUserList.add(new UserListItem(user));
```
Pre-existing problem.

ezaquarii commented on src/main/java/com/owncloud/android/ui/activity/ManageAccountsActivity.java:
```diff
@@ -223,25 +232,20 @@ private void initializeComponentGetters() {
+        for (User user : users) {
+            boolean pendingForRemoval = arbitraryDataProvider.getBooleanValue(user.toPlatformAccount(),
+                                                                              PENDING_FOR_REMOVAL);
+            userListItems.add(new UserListItem(user, !pendingForRemoval));
```
Pre-existing problem.

ezaquarii commented on src/main/java/com/owncloud/android/ui/activity/ManageAccountsActivity.java:
```diff
@@ -155,9 +155,12 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-                accountName = account.name;
-                performAccountRemoval(account);
+                final Account account = Parcels.unwrap(bundle.getParcelable(UserInfoActivity.KEY_ACCOUNT));
+                if (account != null) {
```
Fixed potential NPE. Although this should be refactored further, I don't have available bandwidth to test any more invasive changes.

tobiasKaminsky commented:
Needs a rebase :-)

ezaquarii commented on src/main/java/com/owncloud/android/ui/adapter/UserListAdapter.java:
```diff
@@ -1,449 +1,339 @@
 /*
```
DOH! Gihub is diffing wrong files!
This should be diffed against old `AccountListAdapter`, that has become `UserListAdapter`, becuase it shows users.
Old `UserListAdapter` has become `ShareeListAdapter`, bacuse it shows users sharing a file.

ezaquarii commented:
@tobiasKaminsky rebase failed becuase github diffed wrong files, creation un-reviewable diff. I'll try to fix it somehow.

ezaquarii commented on src/main/java/com/owncloud/android/ui/adapter/ShareeListAdapter.java:
```diff
@@ -0,0 +1,453 @@
+/*
```
This was `UserListAdapter`, renamed into `Sharee*`. Github is confused.

ezaquarii commented on src/main/java/com/owncloud/android/ui/activity/ManageAccountsActivity.java:
```diff
@@ -223,25 +232,20 @@ private void initializeComponentGetters() {
+        for (User user : users) {
+            boolean pendingForRemoval = arbitraryDataProvider.getBooleanValue(user.toPlatformAccount(),
+                                                                              PENDING_FOR_REMOVAL);
+            userListItems.add(new UserListItem(user, !pendingForRemoval));
```
pre-existing issue

ezaquarii commented on src/main/java/com/owncloud/android/ui/dialog/MultipleAccountsDialog.java:
```diff
@@ -117,21 +120,21 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
+        List<User> users = accountManager.getAllUsers();
+        List<UserListItem> adapterUserList = new ArrayList<>(users.size());
+        for (User user : users) {
+            adapterUserList.add(new UserListItem(user));
```
pre-existing issue

ezaquarii commented on src/main/java/com/owncloud/android/ui/activity/ManageAccountsActivity.java:
```diff
@@ -155,9 +155,12 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-                accountName = account.name;
-                performAccountRemoval(account);
+                final Account account = Parcels.unwrap(bundle.getParcelable(UserInfoActivity.KEY_ACCOUNT));
+                if (account != null) {
```
pre-existing issue

tobiasKaminsky commented on src/main/java/com/owncloud/android/ui/activity/UserInfoActivity.java:
```diff
@@ -116,15 +118,22 @@
+        final Account account = Parcels.unwrap(bundle.getParcelable(KEY_ACCOUNT));
+        Optional<User> optionalUser = accountManager.getUser(account != null ? account.name : "");
+        if(!optionalUser.isPresent()) {
+            finish();
```
This would lead to a direct return after clicking on the user/account.
As far as I know, we had an empty for this previously.
Can we have this? 
I can write a test case for this.

tobiasKaminsky commented:
BUG:
```
-02-11 09:48:32.869 6127-6127/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.nextcloud.client, PID: 6127
    org.parceler.ParcelerRuntimeException: Unable to find generated Parcelable class for com.nextcloud.client.account.RegisteredUser, verify that your class is configured properly and that the Parcelable class com.nextcloud.client.account.RegisteredUser$$Parcelable is generated by Parceler.
        at org.parceler.Parcels$ParcelCodeRepository.get(Parcels.java:154)
```

This happens when I click on manage accounts -> and then on one account

ezaquarii commented:
> This happens when I click on manage accounts -> and then on one account

It does not use `Parceler`.

ezaquarii commented on src/main/java/com/owncloud/android/ui/dialog/MultipleAccountsDialog.java:
```diff
@@ -117,21 +120,21 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
+        List<User> users = accountManager.getAllUsers();
+        List<UserListItem> adapterUserList = new ArrayList<>(users.size());
+        for (User user : users) {
+            adapterUserList.add(new UserListItem(user));
```
pre-existing issue

ezaquarii commented on src/main/java/com/owncloud/android/ui/activity/ManageAccountsActivity.java:
```diff
@@ -155,9 +155,12 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-                accountName = account.name;
-                performAccountRemoval(account);
+                final Account account = Parcels.unwrap(bundle.getParcelable(UserInfoActivity.KEY_ACCOUNT));
+                if (account != null) {
```
pre-existing issue

ezaquarii commented on src/main/java/com/owncloud/android/ui/activity/ManageAccountsActivity.java:
```diff
@@ -223,25 +232,20 @@ private void initializeComponentGetters() {
+        for (User user : users) {
+            boolean pendingForRemoval = arbitraryDataProvider.getBooleanValue(user.toPlatformAccount(),
+                                                                              PENDING_FOR_REMOVAL);
+            userListItems.add(new UserListItem(user, !pendingForRemoval));
```
pre-existing issue

ezaquarii commented:
@tobiasKaminsky Rebased. Codacy issues are pre-existing conditions.

tobiasKaminsky commented:
> > This happens when I click on manage accounts -> and then on one account
> 
> It does not use `Parceler`.

It was used here: 
https://github.com/nextcloud/android/pull/5434/files#diff-b380e277786424d15990f349e4d507e0R463

I fixed it and added some screenshot tests.

ezaquarii commented:
This was probably a bug caused by automatic refactoring, when I mass-changed all accounts to users.

On February 12, 2020 8:37:00 AM UTC, Tobias Kaminsky <notifications@github.com> wrote:
>
>
>
>
>> > This happens when I click on manage accounts -> and then on one
>account
>> 
>> It does not use `Parceler`.
>
>It was used here: 
>https://github.com/nextcloud/android/pull/5434/files#diff-b380e277786424d15990f349e4d507e0R463
>
>I fixed it and added some screenshot tests.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/nextcloud/android/pull/5434#issuecomment-585092323
